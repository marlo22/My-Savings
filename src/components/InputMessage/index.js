import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const InputMessageText = styled.Text`
  color: red;
  align-self: flex-end;
  margin-right: 5px;
`;

const InputMessage = ({ message }) => {
  if (!message) {
    return null;
  }

  return (
    <InputMessageText>
      {message}
    </InputMessageText>
  );
};

InputMessage.propTypes = {
  message: PropTypes.string
};

export default InputMessage;