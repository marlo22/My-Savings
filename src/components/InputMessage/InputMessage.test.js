import React from 'react';
import { shallow } from 'enzyme';
import InputMessage from './';

const message = 'Lorem ipsum';

const component = shallow(
  <InputMessage message={message} />
);

describe('<InputMessage />', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should print passed message', () => {
    expect(component.text()).toBe(message);
  });

  test('without message prop should return null', () => {
    component.setProps({ message: null });

    expect(component.html()).toBe(null);
  });
});