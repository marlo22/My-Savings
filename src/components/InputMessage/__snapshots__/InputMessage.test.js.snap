// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputMessage /> snapshot 1`] = `
<Styled(Text)>
  Lorem ipsum
</Styled(Text)>
`;
